src/ledger/LedgerTxn.cpp

```diff
@@ -263,6 +287,35 @@ LedgerTxn::Impl::create(LedgerTxn& self, LedgerEntry const& entry)
     return ltxe;
 }
 
+void
+LedgerTxn::createOrUpdateWithoutLoading(LedgerEntry const& entry)
+{
+    return getImpl()->createOrUpdateWithoutLoading(*this, entry);
+}
+
+void
+LedgerTxn::Impl::createOrUpdateWithoutLoading(LedgerTxn& self,
+                                              LedgerEntry const& entry)
+{
+    throwIfSealed();
+    throwIfChild();
+
+    auto key = LedgerEntryKey(entry);
```

Comment:
Hmm, dangling in the memory-unsafe sense? I thought the point of ltxe being a weak_ptr was that this would (correctly) trap when you try to access ltxe.current()?

Reply:
discussed in-chat, resolved.